core/src/avm2/amf.rs

```diff
@@ -118,31 +121,56 @@ pub fn serialize_value<'gc>(
             } else if let Some(bytearray) = o.as_bytearray() {
                 Some(AmfValue::ByteArray(bytearray.bytes().to_vec()))
             } else {
-                let is_object = o
-                    .instance_of()
-                    .map_or(false, |c| c == activation.avm2().classes().object);
-                if is_object {
-                    let mut object_body = Vec::new();
-                    recursive_serialize(activation, o, &mut object_body, amf_version).unwrap();
-                    Some(AmfValue::Object(
-                        object_body,
-                        if amf_version == AMFVersion::AMF3 {
-                            Some(ClassDefinition {
-                                name: "".to_string(),
-                                attributes: EnumSet::only(Attribute::Dynamic),
-                                static_properties: Vec::new(),
-                            })
-                        } else {
-                            None
-                        },
-                    ))
+                let class = o.instance_of().expect("Missing ClassObject");
+                let qname = QName::new(activation.avm2().flash_net_internal, "_getAliasByClass");
+                let method = activation
+                    .avm2()
+                    .playerglobals_domain
+                    .get_defined_value(activation, qname)
+                    .expect("Failed to lookup flash.net._getAliasByClass");
```

Comment:
This feels like a hack- could you instead add `getAliasByClass` into the AVM2 `SystemClasses`?

Reply:
Unfortunately, I can't - this is a package level function, not class-level